<div class="alert alert-block alert-info"><b>Import</b></div>

In [1]:
from sklearn import ( 
    preprocessing, 
    decomposition,
    )
import pandas as pd 
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns 
import import_ipynb
%run functions.ipynb

<div class="alert alert-block alert-info"><b>Chargement des données</b></div>

In [2]:
loadingPath = "../CSGO-dataset" #chemin où se trouve les quatre fichier .csv avec les données initiales
economy = open(loadingPath+"/economy.csv",encoding="utf8") #table qui resume l'economy des deux equipes sur un match par round
picks = open(loadingPath+"/picks.csv",encoding="utf8") #table qui resume le choix des maps tour a tour des deux equipes et celle restante 
players = open(loadingPath+"/players.csv",encoding="utf8") #table players
results = open(loadingPath+"/results.csv",encoding="utf8") #table results

df_economy = pd.read_csv(economy, low_memory=False)
df_picks = pd.read_csv(picks)
df_players = pd.read_csv(players)
df_results = pd.read_csv(results)

<div class="alert alert-block alert-info"><b>Nouvelles colonnes dans economy et players</b></div>

In [3]:
#res1 = functions.mean_economy_match(df_economy)
#res1.to_csv('economy.csv', index=False)
#res2 = functions.players_to_match(df_players)
#res2.to_csv('players.csv',index=False)

<div class="alert alert-block alert-info"><b>Sélection des colonnes</b></div>

In [21]:
df_read_economy = pd.read_csv('economy.csv',low_memory=False)
# [{0:"date"},{1:"match_id"},{2:"event_id"},{3:"team_1"},{4:"team_2"},{5:"_map"},{6:"economy_1"},{7:"economy_2"}] 
df_reduce_economy = df_read_economy[df_read_economy.columns[[0,1,2,3,4,6,99,100]]] 
# [{0:"date"},{1:"team_1"},{2:"team_2"},{3:"_map"},{4:"result_1"},{5:"result_2"},{6:"starting_ct"},{7:"event_id"},{8:"match_id"},{9:"rank_1"},{10:"rank_2"},{11:"match_winner"}] 
df_reduce_results = df_results[df_results.columns[[0,1,2,3,8,10,12,13,14,15,18]]] 
# [{0:"date"},{1:"event_id"},{2:"match_id"},{3:"team_1"},{4:"team_2"},{5:"country_1"},{6:"country_2"},{7:"kda_1"},{8:"kda_2"},{9:"hs_1"},{10:"hs_2"}] 
df_reduce_players = pd.read_csv('players.csv') 
df_reduce_picks = df_picks

<div class="alert alert-block alert-info"><b>Union des tables</b></div>

In [30]:
df_merge1 = pd.merge(df_reduce_results, df_reduce_economy, on=["team_1","team_2","date","match_id","event_id","_map"], how="inner").drop_duplicates()
df_merge2 = pd.merge(df_merge1,df_reduce_players,on=["team_1","team_2","date","match_id","event_id"],how="inner").drop_duplicates()
df_merge2['date'] = df_merge2['date'].apply(lambda date: int(date[0:4]))
""" [   {0:"date"},{1:"team_1"},{2:"team_2"},{3:"_map"},

        {4:"result_1"},{5:"result_2"},{6:"starting_ct"},
        
        {7:"rank_1"},{8:"rank_2"},{9:"match_winner "},
        
        {10:"economy_1"},{11:"economy_2"},{12:"country_1"},
        
        {13:"country_2"},{14:"kda_1"},{15:"kda_2"},{16:"hs_1"},{17:"hs_2"}] 
        
"""
df_merge2 = df_merge2[df_merge2.columns[[0,3,4,5,6,7,8,9,10,11,12,15,16,17,18]]]
df_merge2

date         _map  ct_1  t_1  event_id  match_id  rank_1  rank_2  \
0      2020        Dust2     8    3      5151   2339817     128      99   
1      2020        Dust2     9    7      5151   2339768      74     143   
2      2020      Vertigo     6   10      5151   2339766      74     128   
3      2020      Inferno     9    7      5152   2339712     188     191   
4      2020      Inferno     7    9      4901   2339388      10      29   
...     ...          ...   ...  ...       ...       ...     ...     ...   
15380  2017       Mirage     7    6      2683   2309417      11       1   
15381  2017  Cobblestone     8    8      2683   2309395       9      33   
15382  2017      Inferno     5    3      2683   2309264       9      11   
15383  2017      Inferno     4    2      2683   2309265      12      16   
15384  2017      Inferno     4   12      2683   2309266       6      82   

       match_winner    economy_1    economy_2     kda_1     kda_2  hs_1  hs_2  
0                 2  3832.142857  4104.285714  1.134086  1.570140   9.0   9.6  
1                 1  4369.230769  3270.000000  1.822549  0.995397   9.6   6.8  
2                 1  4220.400000  3407.600000  1.889167  0.890000   9.6   6.2  
3                 1  3911.785714  3835.714286  1.500082  1.243772   7.0   8.0  
4                 1  4319.230769  3536.923077  1.870168  0.870621  18.8  14.6  
...             ...          ...          ...       ...       ...   ...   ...  
15380             2  3659.333333  3993.333333  1.011996  1.541468   7.4   9.8  
15381             1  3801.000000  3540.333333  1.438655  1.010743   7.4   6.4  
15382             2  3205.200000  4014.800000  0.932138  1.655641   6.8  11.2  
15383             2  3184.782609  4386.086957  0.825460  2.057431   6.0   6.0  
15384             1  3825.384615  3271.923077  1.562949  0.867619  10.6   9.2  

[15385 rows x 15 columns]

<div class="alert alert-block alert-info"><b>Ratio des colonnes</b></div>

In [130]:
#df_expo = ratio_result(df_merge2)
#df_expo.to_csv('ratio_df.csv',index=False)

In [131]:
df_expo = pd.read_csv('ratio_df.csv') 
df_expo

date         _map  result_p1   economy       rank       kda  \
0      2020.0        Dust2   2.666667  0.933693   1.292929  0.722283   
1      2020.0        Dust2   1.285714  1.336156   0.517483  1.830977   
2      2020.0      Vertigo   0.600000  1.238526   0.578125  2.122659   
3      2020.0      Inferno   1.285714  1.019832   0.984293  1.206075   
4      2020.0      Inferno   0.777778  1.221183   0.344828  2.148085   
...       ...          ...        ...       ...        ...       ...   
15380  2017.0       Mirage   1.166667  0.916361  11.000000  0.656514   
15381  2017.0  Cobblestone   1.000000  1.073628   0.272727  1.423365   
15382  2017.0      Inferno   1.666667  0.798346   0.818182  0.563007   
15383  2017.0      Inferno   2.000000  0.726110   0.750000  0.401209   
15384  2017.0      Inferno   0.333333  1.169155   0.073171  1.801423   

       match_winner        hs  
0               2.0  0.937500  
1               1.0  1.411765  
2               1.0  1.548387  
3               1.0  0.875000  
4               1.0  1.287671  
...             ...       ...  
15380           2.0  0.755102  
15381           1.0  1.156250  
15382           2.0  0.607143  
15383           2.0  1.000000  
15384           1.0  1.152174  

[15385 rows x 8 columns]

<div class="alert alert-block alert-info"><b>Filtrage des NaN </b></div>

In [132]:
df_merge2.columns[df_merge2.isnull().any()]

Index([], dtype='object')

In [133]:
#df_merge2['country_2'].isnull().sum()

In [134]:
df_merge2.shape

(15385, 15)

In [135]:
index_with_nan = df_merge2.index[df_merge2.isnull().any(axis=1)]
index_with_nan.shape
df_merge2.drop(index_with_nan, inplace=True)
df_merge2.shape

(15385, 15)

In [136]:
#df_merge2['country_2'].isnull().sum()

<div class="alert alert-block alert-info"><b>Vérification teams wins</b></div>

In [137]:
df_merge2["match_winner"].value_counts(dropna=True) #quasi équitable aussi pour chaque année

1    8194
2    7191
Name: match_winner, dtype: int64

<div class="alert alert-block alert-info"><b>Encodage</b></div>

<div class="alert alert-block alert-warning"><b>Ordinal</b></div>

In [138]:
df_categoricals = df_merge2 .select_dtypes(exclude=['int64','float64']) #données catégorielles 
df_ordinal = df_merge2[df_merge2.columns[[0,2,3,6,7,8,9,10,11,12,13,14]]] #données numériques
df_ordinal

date  ct_1  t_1  rank_1  rank_2  match_winner    economy_1  \
0      2020     8    3     128      99             2  3832.142857   
1      2020     9    7      74     143             1  4369.230769   
2      2020     6   10      74     128             1  4220.400000   
3      2020     9    7     188     191             1  3911.785714   
4      2020     7    9      10      29             1  4319.230769   
...     ...   ...  ...     ...     ...           ...          ...   
15380  2017     7    6      11       1             2  3659.333333   
15381  2017     8    8       9      33             1  3801.000000   
15382  2017     5    3       9      11             2  3205.200000   
15383  2017     4    2      12      16             2  3184.782609   
15384  2017     4   12       6      82             1  3825.384615   

         economy_2     kda_1     kda_2  hs_1  hs_2  
0      4104.285714  1.134086  1.570140   9.0   9.6  
1      3270.000000  1.822549  0.995397   9.6   6.8  
2      3407.600000  1.889167  0.890000   9.6   6.2  
3      3835.714286  1.500082  1.243772   7.0   8.0  
4      3536.923077  1.870168  0.870621  18.8  14.6  
...            ...       ...       ...   ...   ...  
15380  3993.333333  1.011996  1.541468   7.4   9.8  
15381  3540.333333  1.438655  1.010743   7.4   6.4  
15382  4014.800000  0.932138  1.655641   6.8  11.2  
15383  4386.086957  0.825460  2.057431   6.0   6.0  
15384  3271.923077  1.562949  0.867619  10.6   9.2  

[15385 rows x 12 columns]

In [139]:
ord_enc =preprocessing.OrdinalEncoder()
df_ordinal.loc[:,"_map"] = ord_enc.fit_transform(df_categoricals[["_map"]])
df_ordinal

/tmp/ipykernel_17378/1536755155.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ordinal.loc[:,"_map"] = ord_enc.fit_transform(df_categoricals[["_map"]])


date  ct_1  t_1  rank_1  rank_2  match_winner    economy_1  \
0      2020     8    3     128      99             2  3832.142857   
1      2020     9    7      74     143             1  4369.230769   
2      2020     6   10      74     128             1  4220.400000   
3      2020     9    7     188     191             1  3911.785714   
4      2020     7    9      10      29             1  4319.230769   
...     ...   ...  ...     ...     ...           ...          ...   
15380  2017     7    6      11       1             2  3659.333333   
15381  2017     8    8       9      33             1  3801.000000   
15382  2017     5    3       9      11             2  3205.200000   
15383  2017     4    2      12      16             2  3184.782609   
15384  2017     4   12       6      82             1  3825.384615   

         economy_2     kda_1     kda_2  hs_1  hs_2  _map  
0      4104.285714  1.134086  1.570140   9.0   9.6   3.0  
1      3270.000000  1.822549  0.995397   9.6   6.8   3.0  
2      3407.600000  1.889167  0.890000   9.6   6.2   9.0  
3      3835.714286  1.500082  1.243772   7.0   8.0   4.0  
4      3536.923077  1.870168  0.870621  18.8  14.6   4.0  
...            ...       ...       ...   ...   ...   ...  
15380  3993.333333  1.011996  1.541468   7.4   9.8   5.0  
15381  3540.333333  1.438655  1.010743   7.4   6.4   1.0  
15382  4014.800000  0.932138  1.655641   6.8  11.2   4.0  
15383  4386.086957  0.825460  2.057431   6.0   6.0   4.0  
15384  3271.923077  1.562949  0.867619  10.6   9.2   4.0  

[15385 rows x 13 columns]

<div class="alert alert-block alert-warning"><b>One-hot Vector</b></div>

In [140]:
df_one_hot = df_merge2[df_merge2.columns[[0,2,3,6,7,8,9,10,11,12,13,14]]]
df_one_hot

date  ct_1  t_1  rank_1  rank_2  match_winner    economy_1  \
0      2020     8    3     128      99             2  3832.142857   
1      2020     9    7      74     143             1  4369.230769   
2      2020     6   10      74     128             1  4220.400000   
3      2020     9    7     188     191             1  3911.785714   
4      2020     7    9      10      29             1  4319.230769   
...     ...   ...  ...     ...     ...           ...          ...   
15380  2017     7    6      11       1             2  3659.333333   
15381  2017     8    8       9      33             1  3801.000000   
15382  2017     5    3       9      11             2  3205.200000   
15383  2017     4    2      12      16             2  3184.782609   
15384  2017     4   12       6      82             1  3825.384615   

         economy_2     kda_1     kda_2  hs_1  hs_2  
0      4104.285714  1.134086  1.570140   9.0   9.6  
1      3270.000000  1.822549  0.995397   9.6   6.8  
2      3407.600000  1.889167  0.890000   9.6   6.2  
3      3835.714286  1.500082  1.243772   7.0   8.0  
4      3536.923077  1.870168  0.870621  18.8  14.6  
...            ...       ...       ...   ...   ...  
15380  3993.333333  1.011996  1.541468   7.4   9.8  
15381  3540.333333  1.438655  1.010743   7.4   6.4  
15382  4014.800000  0.932138  1.655641   6.8  11.2  
15383  4386.086957  0.825460  2.057431   6.0   6.0  
15384  3271.923077  1.562949  0.867619  10.6   9.2  

[15385 rows x 12 columns]

In [141]:
for num_col in range(df_categoricals.shape[1]):
    df_one_hot_name_col = pd.get_dummies(df_categoricals[df_categoricals.columns[num_col]], prefix=df_categoricals.columns[num_col])
    df_one_hot = pd.concat([df_one_hot,df_one_hot_name_col],axis=1, join="inner")

df_one_hot

date  ct_1  t_1  rank_1  rank_2  match_winner    economy_1  \
0      2020     8    3     128      99             2  3832.142857   
1      2020     9    7      74     143             1  4369.230769   
2      2020     6   10      74     128             1  4220.400000   
3      2020     9    7     188     191             1  3911.785714   
4      2020     7    9      10      29             1  4319.230769   
...     ...   ...  ...     ...     ...           ...          ...   
15380  2017     7    6      11       1             2  3659.333333   
15381  2017     8    8       9      33             1  3801.000000   
15382  2017     5    3       9      11             2  3205.200000   
15383  2017     4    2      12      16             2  3184.782609   
15384  2017     4   12       6      82             1  3825.384615   

         economy_2     kda_1     kda_2  ...  _map_Cache  _map_Cobblestone  \
0      4104.285714  1.134086  1.570140  ...           0                 0   
1      3270.000000  1.822549  0.995397  ...           0                 0   
2      3407.600000  1.889167  0.890000  ...           0                 0   
3      3835.714286  1.500082  1.243772  ...           0                 0   
4      3536.923077  1.870168  0.870621  ...           0                 0   
...            ...       ...       ...  ...         ...               ...   
15380  3993.333333  1.011996  1.541468  ...           0                 0   
15381  3540.333333  1.438655  1.010743  ...           0                 1   
15382  4014.800000  0.932138  1.655641  ...           0                 0   
15383  4386.086957  0.825460  2.057431  ...           0                 0   
15384  3271.923077  1.562949  0.867619  ...           0                 0   

       _map_Default  _map_Dust2  _map_Inferno  _map_Mirage  _map_Nuke  \
0                 0           1             0            0          0   
1                 0           1             0            0          0   
2                 0           0             0            0          0   
3                 0           0             1            0          0   
4                 0           0             1            0          0   
...             ...         ...           ...          ...        ...   
15380             0           0             0            1          0   
15381             0           0             0            0          0   
15382             0           0             1            0          0   
15383             0           0             1            0          0   
15384             0           0             1            0          0   

       _map_Overpass  _map_Train  _map_Vertigo  
0                  0           0             0  
1                  0           0             0  
2                  0           0             1  
3                  0           0             0  
4                  0           0             0  
...              ...         ...           ...  
15380              0           0             0  
15381              0           0             0  
15382              0           0             0  
15383              0           0             0  
15384              0           0             0  

[15385 rows x 22 columns]

In [142]:
df_oh = df_expo[df_expo.columns[[0,2,3,4,5,6,7]]]
df_cat = df_expo.select_dtypes(exclude=['int64','float64'])


for num_col in range(df_cat.shape[1]):
    df_oh_name_col = pd.get_dummies(df_cat[df_cat.columns[num_col]], prefix=df_cat.columns[num_col])
    df_oh = pd.concat([df_oh,df_oh_name_col],axis=1, join="inner")
    
df_oh

date  result_p1   economy       rank       kda  match_winner  \
0      2020.0   2.666667  0.933693   1.292929  0.722283           2.0   
1      2020.0   1.285714  1.336156   0.517483  1.830977           1.0   
2      2020.0   0.600000  1.238526   0.578125  2.122659           1.0   
3      2020.0   1.285714  1.019832   0.984293  1.206075           1.0   
4      2020.0   0.777778  1.221183   0.344828  2.148085           1.0   
...       ...        ...       ...        ...       ...           ...   
15380  2017.0   1.166667  0.916361  11.000000  0.656514           2.0   
15381  2017.0   1.000000  1.073628   0.272727  1.423365           1.0   
15382  2017.0   1.666667  0.798346   0.818182  0.563007           2.0   
15383  2017.0   2.000000  0.726110   0.750000  0.401209           2.0   
15384  2017.0   0.333333  1.169155   0.073171  1.801423           1.0   

             hs  _map_Cache  _map_Cobblestone  _map_Default  _map_Dust2  \
0      0.937500           0                 0             0           1   
1      1.411765           0                 0             0           1   
2      1.548387           0                 0             0           0   
3      0.875000           0                 0             0           0   
4      1.287671           0                 0             0           0   
...         ...         ...               ...           ...         ...   
15380  0.755102           0                 0             0           0   
15381  1.156250           0                 1             0           0   
15382  0.607143           0                 0             0           0   
15383  1.000000           0                 0             0           0   
15384  1.152174           0                 0             0           0   

       _map_Inferno  _map_Mirage  _map_Nuke  _map_Overpass  _map_Train  \
0                 0            0          0              0           0   
1                 0            0          0              0           0   
2                 0            0          0              0           0   
3                 1            0          0              0           0   
4                 1            0          0              0           0   
...             ...          ...        ...            ...         ...   
15380             0            1          0              0           0   
15381             0            0          0              0           0   
15382             1            0          0              0           0   
15383             1            0          0              0           0   
15384             1            0          0              0           0   

       _map_Vertigo  
0                 0  
1                 0  
2                 1  
3                 0  
4                 0  
...             ...  
15380             0  
15381             0  
15382             0  
15383             0  
15384             0  

[15385 rows x 17 columns]

<div class="alert alert-block alert-info"><b>Mean standardisation</b></div>

In [196]:
def standardize_zscore(df):
    '''Standardize all the columns except the last one (target values).'''
    df_scaled = (df-df.mean())/df.std()
    df_scaled.iloc[:,0] = df.iloc[:,0]
    df_scaled.iloc[:,5] = df.iloc[:,5]
    return df_scaled

<div class="alert alert-block alert-warning"><b>Standardisation Ordinal</b></div>

In [197]:
df_ordinal_standard = standardize_zscore(df_ordinal)
df_ordinal_standard

date      ct_1       t_1    rank_1    rank_2  match_winner  economy_1  \
0      2020  0.428361 -1.062967  0.930182  0.296259             2  -0.081600   
1      2020  0.735846  0.204765  0.062120  0.965329             1   1.063105   
2      2020 -0.186610  1.155565  0.062120  0.737237             1   0.745899   
3      2020  0.735846  0.204765  1.894694  1.695223             1   0.088144   
4      2020  0.120876  0.838632 -0.966693 -0.768170             1   0.956539   
...     ...       ...       ...       ...       ...           ...        ...   
15380  2017  0.120876 -0.112168 -0.950618 -1.193941             2  -0.449912   
15381  2017  0.428361  0.521699 -0.982768 -0.707345             1  -0.147976   
15382  2017 -0.494095 -1.062967 -0.982768 -1.041880             2  -1.417815   
15383  2017 -0.801580 -1.379901 -0.934543 -0.965850             2  -1.461331   
15384  2017 -0.801580  1.789431 -1.030994  0.037755             1  -0.096004   

       economy_2     kda_1     kda_2      hs_1      hs_2      _map  
0       0.666249 -0.503106  0.754190 -1.029647 -0.912828 -0.795152  
1      -1.073338  1.198028 -0.733727 -0.947003 -1.296155 -0.795152  
2      -0.786425  1.362634 -1.006582 -0.947003 -1.378296  1.742549  
3       0.106245  0.401239 -0.090724 -1.305130 -1.131872 -0.372202  
4      -0.516771  1.315690 -1.056751  0.320215 -0.228316 -0.372202  
...          ...       ...       ...       ...       ...       ...  
15380   0.434900 -0.804781  0.679963 -1.250033 -0.885447  0.050749  
15381  -0.509660  0.249459 -0.694000 -1.250033 -1.350915 -1.641052  
15382   0.479661 -1.002102  0.975538 -1.332678 -0.693784 -0.372202  
15383   1.253839 -1.265695  2.015708 -1.442871 -1.405676 -0.372202  
15384  -1.069328  0.556577 -1.064523 -0.809262 -0.967589 -0.372202  

[15385 rows x 13 columns]

<div class="alert alert-block alert-warning"><b>Standardisation One Hot</b></div>

In [198]:
df_one_hot_standard = standardize_zscore(df_one_hot)
df_one_hot_standard

date      ct_1       t_1    rank_1    rank_2  match_winner  economy_1  \
0      2020  0.428361 -1.062967  0.930182  0.296259             2  -0.081600   
1      2020  0.735846  0.204765  0.062120  0.965329             1   1.063105   
2      2020 -0.186610  1.155565  0.062120  0.737237             1   0.745899   
3      2020  0.735846  0.204765  1.894694  1.695223             1   0.088144   
4      2020  0.120876  0.838632 -0.966693 -0.768170             1   0.956539   
...     ...       ...       ...       ...       ...           ...        ...   
15380  2017  0.120876 -0.112168 -0.950618 -1.193941             2  -0.449912   
15381  2017  0.428361  0.521699 -0.982768 -0.707345             1  -0.147976   
15382  2017 -0.494095 -1.062967 -0.982768 -1.041880             2  -1.417815   
15383  2017 -0.801580 -1.379901 -0.934543 -0.965850             2  -1.461331   
15384  2017 -0.801580  1.789431 -1.030994  0.037755             1  -0.096004   

       economy_2     kda_1     kda_2  ...  _map_Cache  _map_Cobblestone  \
0       0.666249 -0.503106  0.754190  ...   -0.303028         -0.221409   
1      -1.073338  1.198028 -0.733727  ...   -0.303028         -0.221409   
2      -0.786425  1.362634 -1.006582  ...   -0.303028         -0.221409   
3       0.106245  0.401239 -0.090724  ...   -0.303028         -0.221409   
4      -0.516771  1.315690 -1.056751  ...   -0.303028         -0.221409   
...          ...       ...       ...  ...         ...               ...   
15380   0.434900 -0.804781  0.679963  ...   -0.303028         -0.221409   
15381  -0.509660  0.249459 -0.694000  ...   -0.303028          4.516243   
15382   0.479661 -1.002102  0.975538  ...   -0.303028         -0.221409   
15383   1.253839 -1.265695  2.015708  ...   -0.303028         -0.221409   
15384  -1.069328  0.556577 -1.064523  ...   -0.303028         -0.221409   

       _map_Default  _map_Dust2  _map_Inferno  _map_Mirage  _map_Nuke  \
0         -0.019751    3.232104     -0.495613    -0.492661  -0.332781   
1         -0.019751    3.232104     -0.495613    -0.492661  -0.332781   
2         -0.019751   -0.309376     -0.495613    -0.492661  -0.332781   
3         -0.019751   -0.309376      2.017573    -0.492661  -0.332781   
4         -0.019751   -0.309376      2.017573    -0.492661  -0.332781   
...             ...         ...           ...          ...        ...   
15380     -0.019751   -0.309376     -0.495613     2.029663  -0.332781   
15381     -0.019751   -0.309376     -0.495613    -0.492661  -0.332781   
15382     -0.019751   -0.309376      2.017573    -0.492661  -0.332781   
15383     -0.019751   -0.309376      2.017573    -0.492661  -0.332781   
15384     -0.019751   -0.309376      2.017573    -0.492661  -0.332781   

       _map_Overpass  _map_Train  _map_Vertigo  
0          -0.376139   -0.416665     -0.132637  
1          -0.376139   -0.416665     -0.132637  
2          -0.376139   -0.416665      7.538876  
3          -0.376139   -0.416665     -0.132637  
4          -0.376139   -0.416665     -0.132637  
...              ...         ...           ...  
15380      -0.376139   -0.416665     -0.132637  
15381      -0.376139   -0.416665     -0.132637  
15382      -0.376139   -0.416665     -0.132637  
15383      -0.376139   -0.416665     -0.132637  
15384      -0.376139   -0.416665     -0.132637  

[15385 rows x 22 columns]

<div class="alert alert-block alert-warning"><b>Standardisation One Hot Ratio</b></div>

In [203]:
df_oh_std = standardize_zscore(df_oh)
df_oh_std

date  result_p1   economy      rank       kda  match_winner  \
0      2020.0   0.785349 -0.477895 -0.017910 -0.607417           2.0   
1      2020.0  -0.091369  1.166700 -0.332843  0.715374           1.0   
2      2020.0  -0.526705  0.767750 -0.308214  1.063382           1.0   
3      2020.0  -0.091369 -0.125901 -0.143256 -0.030201           1.0   
4      2020.0  -0.413840  0.696883 -0.402963  1.093718           1.0   
...       ...        ...       ...       ...       ...           ...   
15380  2017.0  -0.166948 -0.548720  3.924429 -0.685886           2.0   
15381  2017.0  -0.272759  0.093924 -0.432245  0.229049           1.0   
15382  2017.0   0.150484 -1.030966 -0.210719 -0.797450           2.0   
15383  2017.0   0.362106 -1.326145 -0.238410 -0.990493           2.0   
15384  2017.0  -0.696002  0.484278 -0.513291  0.680113           1.0   

             hs  _map_Cache  _map_Cobblestone  _map_Default  _map_Dust2  \
0     -0.380111   -0.303028         -0.221409     -0.019751    3.232104   
1      1.085839   -0.303028         -0.221409     -0.019751    3.232104   
2      1.508138   -0.303028         -0.221409     -0.019751   -0.309376   
3     -0.573299   -0.303028         -0.221409     -0.019751   -0.309376   
4      0.702266   -0.303028         -0.221409     -0.019751   -0.309376   
...         ...         ...               ...           ...         ...   
15380 -0.943903   -0.303028         -0.221409     -0.019751   -0.309376   
15381  0.296044   -0.303028          4.516243     -0.019751   -0.309376   
15382 -1.401244   -0.303028         -0.221409     -0.019751   -0.309376   
15383 -0.186924   -0.303028         -0.221409     -0.019751   -0.309376   
15384  0.283445   -0.303028         -0.221409     -0.019751   -0.309376   

       _map_Inferno  _map_Mirage  _map_Nuke  _map_Overpass  _map_Train  \
0         -0.495613    -0.492661  -0.332781      -0.376139   -0.416665   
1         -0.495613    -0.492661  -0.332781      -0.376139   -0.416665   
2         -0.495613    -0.492661  -0.332781      -0.376139   -0.416665   
3          2.017573    -0.492661  -0.332781      -0.376139   -0.416665   
4          2.017573    -0.492661  -0.332781      -0.376139   -0.416665   
...             ...          ...        ...            ...         ...   
15380     -0.495613     2.029663  -0.332781      -0.376139   -0.416665   
15381     -0.495613    -0.492661  -0.332781      -0.376139   -0.416665   
15382      2.017573    -0.492661  -0.332781      -0.376139   -0.416665   
15383      2.017573    -0.492661  -0.332781      -0.376139   -0.416665   
15384      2.017573    -0.492661  -0.332781      -0.376139   -0.416665   

       _map_Vertigo  
0         -0.132637  
1         -0.132637  
2          7.538876  
3         -0.132637  
4         -0.132637  
...             ...  
15380     -0.132637  
15381     -0.132637  
15382     -0.132637  
15383     -0.132637  
15384     -0.132637  

[15385 rows x 17 columns]

<div class="alert alert-block alert-info"><b>Correlation Map</b></div>

In [205]:
corrMatrix_ordinal = df_expo.corr()
#plt.figure(figsize=(30,16))
#sns.heatmap(corrMatrix_ordinal, annot=True,mask=np.triu(corrMatrix_ordinal))
#plt.title("Map de corrélation des données avec ratio")
#plt.show()

In [206]:
corrMatrix_ordinal = df_merge2.corr()
#plt.figure(figsize=(30,16))
#plt.title("Map de corrélation des données brutes")
#sns.heatmap(corrMatrix_ordinal, annot=True,mask=np.triu(corrMatrix_ordinal))
#plt.show()

#----------Negative----Positive---
#Faible de −0,5 à 0,0 de 0,0 à 0,5   
#Forte de −1,0 à −0,5 de 0,5 à 1,0

In [207]:
corrMatrix_one_hot = df_one_hot.corr()
#plt.figure(figsize=(30,16))
#sns.heatmap(corrMatrix_one_hot, annot=True,cmap="tab20",mask=np.triu(corrMatrix_one_hot))

<div class="alert alert-block alert-info"><b>Classification par date</b></div>

In [208]:
df_2017_ordinal = df_ordinal[df_ordinal["date"] == 2017].reset_index(drop=True)
df_2018_ordinal = df_ordinal[df_ordinal["date"] == 2018].reset_index(drop=True)
df_2019_ordinal = df_ordinal[df_ordinal["date"] == 2019].reset_index(drop=True)
df_2020_ordinal = df_ordinal[df_ordinal["date"] == 2020].reset_index(drop=True)

df_2017_one_hot = df_one_hot[df_one_hot["date"] == 2017].reset_index(drop=True)
df_2018_one_hot = df_one_hot[df_one_hot["date"] == 2018].reset_index(drop=True)
df_2019_one_hot = df_one_hot[df_one_hot["date"] == 2019].reset_index(drop=True)
df_2020_one_hot = df_one_hot[df_one_hot["date"] == 2020].reset_index(drop=True)

df_2017_ordinal_standard = df_ordinal_standard[df_ordinal_standard["date"] == 2017].reset_index(drop=True)
df_2018_ordinal_standard = df_ordinal_standard[df_ordinal_standard["date"] == 2018].reset_index(drop=True)
df_2019_ordinal_standard = df_ordinal_standard[df_ordinal_standard["date"] == 2019].reset_index(drop=True)
df_2020_ordinal_standard = df_ordinal_standard[df_ordinal_standard["date"] == 2020].reset_index(drop=True)

df_2017_one_hot_standard = df_one_hot_standard[df_one_hot_standard["date"] == 2017].reset_index(drop=True)
df_2018_one_hot_standard = df_one_hot_standard[df_one_hot_standard["date"] == 2018].reset_index(drop=True)
df_2019_one_hot_standard = df_one_hot_standard[df_one_hot_standard["date"] == 2019].reset_index(drop=True)
df_2020_one_hot_standard = df_one_hot_standard[df_one_hot_standard["date"] == 2020].reset_index(drop=True)

df_2017_one_hot_ratio_standard = df_oh_std[df_oh_std["date"] == 2017].reset_index(drop=True)
df_2018_one_hot_ratio_standard = df_oh_std[df_oh_std["date"] == 2018].reset_index(drop=True)
df_2019_one_hot_ratio_standard = df_oh_std[df_oh_std["date"] == 2019].reset_index(drop=True)
df_2020_one_hot_ratio_standard = df_oh_std[df_oh_std["date"] == 2020].reset_index(drop=True)

<div class="alert alert-block alert-info"><b>Pourcentage validation, train, test set</b></div>

In [209]:
['merge2 : ' + str(df_merge2.shape[0]),'2017 : ' + str(df_2017_ordinal.shape[0]),'2018 : ' + str(df_2018_ordinal.shape[0]),'2019 : ' + str(df_2019_ordinal.shape[0]),'2020 : ' + str(df_2020_ordinal.shape[0])]

['merge2 : 15385', '2017 : 3263', '2018 : 6272', '2019 : 5260', '2020 : 590']

- train             (80% de 15385 = 12308)           =>         df_2018 + df_2019 + df_2020
- validation        (10% de 15385 = 1538.5)          =>         1631 * df_2017
- test              (10% de 15385 = 1538.5)          =>         1631 * df_2017

In [210]:
train_set_ordinal = pd.concat((df_2018_ordinal,df_2019_ordinal,df_2020_ordinal)).reset_index(drop=True)
validation_set_ordinal = df_2017_ordinal.head(1631).reset_index().reset_index(drop=True)
test_set_ordinal = df_2017_ordinal.drop(df_2017_ordinal.index[:1631]).head(1631).reset_index(drop=True)

train_set_one_hot = pd.concat((df_2018_one_hot,df_2019_one_hot,df_2020_one_hot)).reset_index(drop=True)
validation_one_hot = df_2017_one_hot.head(1631).reset_index(drop=True)
test_set_one_hot = df_2017_one_hot.drop(df_2017_one_hot.index[:1631]).head(1631).reset_index(drop=True)

train_set_ordinal_standard = pd.concat((df_2018_ordinal_standard,df_2019_ordinal_standard,df_2020_ordinal_standard)).reset_index(drop=True)
validation_set_ordinal_standard = df_2017_ordinal_standard.head(1631).reset_index().reset_index(drop=True)
test_set_ordinal_standard = df_2017_ordinal_standard.drop(df_2017_ordinal_standard.index[:1631]).head(1631).reset_index(drop=True)

train_set_one_hot_standard = pd.concat((df_2018_one_hot_standard,df_2019_one_hot_standard,df_2020_one_hot_standard)).reset_index(drop=True)
validation_one_hot_standard = df_2017_one_hot_standard.head(1631).reset_index(drop=True)
test_set_one_hot_standard = df_2017_one_hot_standard.drop(df_2017_one_hot_standard.index[:1631]).head(1631).reset_index(drop=True)

train_set_one_hot_ratio_standard = pd.concat((df_2018_one_hot_ratio_standard,df_2019_one_hot_ratio_standard,df_2020_one_hot_ratio_standard)).reset_index(drop=True)
validation_one_hot_ratio_standard = df_2017_one_hot_ratio_standard.head(1631).reset_index(drop=True)
test_set_one_hot_ratio_standard = df_2017_one_hot_ratio_standard.drop(df_2017_one_hot_ratio_standard.index[:1631]).head(1631).reset_index(drop=True)

In [211]:
x_train_ordinal = pd.concat([train_set_ordinal.iloc[:,:5],train_set_ordinal.iloc[:,6:]],axis=1,join='inner')
y_train_ordinal = train_set_ordinal['match_winner'].reset_index(drop=True)

x_train_one_hot = pd.concat([train_set_one_hot.iloc[:,:5],train_set_one_hot.iloc[:,6:]],axis=1,join='inner')
y_train_one_hot = train_set_one_hot['match_winner'].reset_index(drop=True)

x_train_ordinal_standard = pd.concat([train_set_ordinal_standard.iloc[:,:5],train_set_ordinal_standard.iloc[:,6:]],axis=1,join='inner')
y_train_ordinal_standard = train_set_ordinal_standard['match_winner'].reset_index(drop=True)

x_train_one_hot_standard = pd.concat([train_set_one_hot_standard.iloc[:,:5],train_set_one_hot_standard.iloc[:,6:]],axis=1,join='inner')
y_train_one_hot_standard = train_set_one_hot_standard['match_winner'].reset_index(drop=True)

x_train_one_hot_ratio_standard = pd.concat([train_set_one_hot_ratio_standard.iloc[:,:5],train_set_one_hot_ratio_standard.iloc[:,6:]],axis=1,join='inner')
y_train_one_hot_ratio_standard = train_set_one_hot_ratio_standard['match_winner'].reset_index(drop=True)


In [212]:
x_test_ordinal = pd.concat([test_set_ordinal.iloc[:,:5],test_set_ordinal.iloc[:,6:]],axis=1,join='inner')
y_test_ordinal = test_set_ordinal['match_winner'].reset_index(drop=True)

x_test_one_hot = pd.concat([test_set_one_hot.iloc[:,:5],test_set_one_hot.iloc[:,6:]],axis=1,join='inner')
y_test_one_hot = test_set_one_hot['match_winner'].reset_index(drop=True)

x_test_ordinal_standard = pd.concat([test_set_ordinal_standard.iloc[:,:5],test_set_ordinal_standard.iloc[:,6:]],axis=1,join='inner')
y_test_ordinal_standard = test_set_ordinal_standard['match_winner'].reset_index(drop=True)

x_test_one_hot_standard = pd.concat([test_set_one_hot_standard.iloc[:,:5],test_set_one_hot_standard.iloc[:,6:]],axis=1,join='inner')
y_test_one_hot_standard = test_set_one_hot_standard['match_winner'].reset_index(drop=True)

x_test_one_hot_ratio_standard = pd.concat([test_set_one_hot_ratio_standard.iloc[:,:5],test_set_one_hot_ratio_standard.iloc[:,6:]],axis=1,join='inner')
y_test_one_hot_ratio_standard = test_set_one_hot_ratio_standard['match_winner'].reset_index(drop=True)

<div class="alert alert-block alert-info"><b>PCA</b></div>

<div class="alert alert-block alert-warning"><b>Ordinal</b></div>

In [213]:
from sklearn.decomposition import PCA
pca = decomposition.PCA(n_components=5) #le nombre de components à garder
data_ordinal = pca.fit_transform(df_ordinal) #pca
#print(data_ordinal)
#print(data_ordinal.shape)

<div class="alert alert-block alert-warning"><b>Ordinal Standard</b></div>

In [214]:
data_ordinal_standard = pca.fit_transform(df_ordinal_standard) #pca
#print(data_ordinal_standard)
#print(data_ordinal_standard.shape)

<div class="alert alert-block alert-warning"><b>One-hot vector</b></div>

In [215]:
data_one_hot = pca.fit_transform(df_one_hot) #pca
#print(data_one_hot)
#print(data_one_hot.shape)

<div class="alert alert-block alert-warning"><b>One-hot vector Standard</b></div>

In [216]:
data_one_hot_standard = pca.fit_transform(df_one_hot_standard) #pca
#print(data_one_hot_standard )
#print(data_one_hot_standard.shape)